state/megawatcher.go

```diff
@@ -491,6 +496,86 @@ func backingEntityIdForSettingsKey(key string) (eid multiwatcher.EntityId, extra
 	return
 }
 
+type backingOpenedPorts map[string]interface{}
+
+func (p *backingOpenedPorts) updated(st *State, store *multiwatcherStore, id interface{}) error {
+	localID := st.localID(id.(string))
+	parentId, ok := backingEntityIdForOpenedPortsKey(localID)
+	if !ok {
+		return nil
+	}
+	info0 := store.Get(parentId)
+	switch info := info0.(type) {
+	case nil:
+		// The parent info doesn't exist. This is unexpected because the port
+		// always refers to a machine. Anyway, ignore the ports for now.
+		return nil
+	case *multiwatcher.MachineInfo:
+		// Retrieve the machine.
+		m, err := st.Machine(info.Id)
```

Comment:
It might be worth investigating if we can reasonably avoid this round-trip,
as all Machine.Units actually requires is the machine id, which we already have.